main/src/main/scala/sbt/Defaults.scala

```diff
@@ -739,7 +743,17 @@ object Defaults extends BuildCommon {
   def runnerInit: Initialize[Task[ScalaRun]] = Def.task {
     val tmp = taskTemporaryDirectory.value
     val si = scalaInstance.value
-    if (fork.value) new ForkRun(forkOptions.value) else new Run(si, trapExit.value, tmp)
+    val s = streams.value
+    val options = javaOptions.value
+    if (fork.value) {
+      s.log.debug(s"javaOptions: $options")
+      new ForkRun(forkOptions.value)
+    } else {
+      if (options.nonEmpty) {
+        s.log.warn("javaOptions will be ignored, fork is set to false")
```

Comment:
Also, thinking about this, it's not actually telling WHICH CONFIGURATION has javaOptions and fork set to false.  You may actually want to grab the magic "what's my current configuration" dependency so you can report which one is the problem, otherwise it'll just be popping out and giving the user no idea how to solve it.  GIve me a abit and I'll look up the syntax.